pkg/deployment/merge.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright © 2025 SUSE LLC
+SPDX-License-Identifier: Apache-2.0
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deployment
+
+import (
+	"fmt"
+	"reflect"
+	"slices"
+
+	"dario.cat/mergo"
+)
+
+type transformer struct{}
+
+// Transformer tells mergo which types this transformer is for.
+func (t *transformer) Transformer(typ reflect.Type) func(dest, src reflect.Value) error {
+	// Check if the type is the one we want to customize
+	switch {
+	case typ == reflect.TypeOf([]*Disk{}):
+		return t.mergeDisks()
+	case typ == reflect.TypeOf(Partitions{}):
+		return t.mergePartitions()
+	}
+	return nil
```

Comment:
With this pattern we can easily customize how to merge any specific type. So far only slices required customization to merge them one by one instead of the replacing the whole slice. We can easily grow the custom logic if we detect further corner cases.